### PR TITLE
[DOCS] Added Documentation for Sprite3D, AnimatedSprite3D and SpriteBase3D and updated Sprite and Animatedsprite

### PR DESCRIPTION
**Pull request by Grosskopf**
Mostly copied from Sprite2D and AnimatedSprite2D, added some and failed to document the AlphaCut in SpriteBase3D as i don't understand it and get some Bugs trying it out

### CODE REVIEW

File: doc/classes/AnimatedSprite3D.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AnimatedSprite3D" inherits="SpriteBase3D" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Sprite node in 3D that can use multiple 2D textures for animation.
 	</brief_description>
 	<description>
+		Sprite node in 3D that can use multiple 2D textures for animation. Animations are created using a [SpriteFrames] resource, which can be configured in the editor via the SpriteFrames panel.
```
Comment (NathanLovato) [inline doc/classes/AnimatedSprite3D.xml]: Avoid repeating the brief description: develop it or at least don't duplicate it. Avoid the passive voice. Example:
`Animations rely on a [SpriteFrames] resource and work the same as [AnimatedSprite]: use the SpriteFrames bottom panel in the editor to set them up..`

File: doc/classes/AnimatedSprite3D.xml [hunk 2]
```diff
@@ -83,12 +85,16 @@
 	</methods>
 	<members>
 		<member name="animation" type="String" setter="set_animation" getter="get_animation">
+			The current animation from the [code]frames[/code] resource. If this value is changed, the [code]frame[/code] counter is reset.
```
Comment (NathanLovato) [inline doc/classes/AnimatedSprite3D.xml]: Link to member variables with [member frame], the docs should support this soon.
Passive voice: `If this value is changed` -> `If this value changes`

File: doc/classes/AnimatedSprite3D.xml [hunk 3]
```diff
@@ -83,12 +85,16 @@
...
 		</member>
 		<member name="frame" type="int" setter="set_frame" getter="get_frame">
+			The current frame index.
```
Comment (NathanLovato) [inline doc/classes/AnimatedSprite3D.xml]: Could be clearer. Suggestion: `The index of the animation frame the node displays/displayed on screen.`
NB: passive voice is okay in some instances - when the node is acted upon, here processed by the renderer.

File: doc/classes/AnimatedSprite3D.xml [hunk 4]
```diff
@@ -83,12 +85,16 @@
...
 		</member>
 		<member name="frames" type="SpriteFrames" setter="set_sprite_frames" getter="get_sprite_frames">
+			The [SpriteFrames] resource containing the animation(s).
 		</member>
 		<member name="playing" type="bool" setter="_set_playing" getter="_is_playing">
+			If [code]true[/code] the [code]animation[/code] is currently playing.
```
Comment (NathanLovato) [inline doc/classes/AnimatedSprite3D.xml]: Suggestion: `[code]true[/code] if the [member animation] is currently playing.`

File: doc/classes/Sprite3D.xml [hunk 5]
```diff
@@ -96,21 +98,28 @@
 	</methods>
 	<members>
 		<member name="frame" type="int" setter="set_frame" getter="get_frame">
+			Current frame to display from sprite sheet. [code]vframes[/code] or [code]hframes[/code] must be greater than 1.
 		</member>
 		<member name="hframes" type="int" setter="set_hframes" getter="get_hframes">
+			The number of horizontal frames in the sprite sheet.
 		</member>
 		<member name="region_enabled" type="bool" setter="set_region" getter="is_region">
+			If [code]true[/code] texture will be cut from a larger atlas texture. See [code]region_rect[/code]. Default value: [code]false[/code].
```
Comment (NathanLovato) [inline doc/classes/Sprite3D.xml]: Avoid future too. It adds extra words.
`If [code]true[/code], the node's texture is cut from a texture atlas. See [member region_rect]. Default value: [code]false[/code].

File: doc/classes/Sprite3D.xml [hunk 6]
```diff
@@ -96,21 +98,28 @@
...
 		</member>
 		<member name="region_rect" type="Rect2" setter="set_region_rect" getter="get_region_rect">
+			The region of the atlas texture to display. [code]region_enabled[/code] must be [code]true[/code].
 		</member>
 		<member name="texture" type="Texture" setter="set_texture" getter="get_texture">
+			[Texture] object to draw.
 		</member>
 		<member name="vframes" type="int" setter="set_vframes" getter="get_vframes">
+			The number of vertical frames in the sprite sheet.
```
Comment (NathanLovato) [inline doc/classes/Sprite3D.xml]: What's a vertical frame? Is it a row count?

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Thank you very much for your work. The PR does need a little polishing.
Please review the [writing guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html). In summary:
- Avoid the passive voice
- Avoid future tense
- Use [member member_name] to link to member vars (should work soon)
Ping me when you've amended your commit so I can approve and merge :)

**Review by cbscribe — COMMENTED** [review]
On top of @NathanLovato 's comments - I know you took a bunch of these descriptions from Sprite & AnimatedSprite. If you fix things, can you copy them back over to those classes as well?

File: doc/classes/SpriteBase3D.xml [hunk 7]
```diff
@@ -164,36 +166,51 @@
 		<member name="alpha_cut" type="int" setter="set_alpha_cut_mode" getter="get_alpha_cut_mode" enum="SpriteBase3D.AlphaCutMode">
 		</member>
 		<member name="axis" type="int" setter="set_axis" getter="get_axis" enum="Vector3.Axis">
+			The Direction in wich the front of the texture faces.
 		</member>
 		<member name="centered" type="bool" setter="set_centered" getter="is_centered">
+			If [code]true[/code], texture will be centered. Default value: [code]true[/code].
```
Comment (cbscribe) [inline doc/classes/SpriteBase3D.xml]: The comma here is not necessary. For the rest of the docs we've standardized on no comma after "If true", so please change to follow suit.

File: doc/classes/SpriteBase3D.xml [hunk 8]
```diff
@@ -164,36 +166,51 @@
...
 		<member name="shaded" type="bool" setter="set_draw_flag" getter="get_draw_flag">
+			If [code]true[/code], the Light in the Environment has effects on the Sprite. Default value: [code]false[/code].
 		</member>
 		<member name="transparent" type="bool" setter="set_draw_flag" getter="get_draw_flag">
+			If [code]true[/code], the Texture's transparency and the opacity are used to make those parts of the Sprite invisible. Default value: [code]true[/code].
 		</member>
 	</members>
 	<constants>
 		<constant name="FLAG_TRANSPARENT" value="0">
+			If set, the Texture's transparency and the opacity are used to make those parts of the Sprite invisible.
```
Comment (cbscribe) [inline doc/classes/SpriteBase3D.xml]: You're inconsistent about capitalizing "Texture". Unless you're linking to the Texture class, using `[Texture]`, this should be lowercase.

File: doc/classes/SpriteBase3D.xml [hunk 9]
```diff
@@ -164,36 +166,51 @@
...
+			The texture's drawing offset.
 		</member>
 		<member name="opacity" type="float" setter="set_opacity" getter="get_opacity">
+			The objects visibility on a scale from [code]0[/code] fully invisible to [code]1[/code] fully visible.
 		</member>
 		<member name="pixel_size" type="float" setter="set_pixel_size" getter="get_pixel_size">
+			The size of one pixel's width on the Sprite to scale it in 3D.
 		</member>
 		<member name="shaded" type="bool" setter="set_draw_flag" getter="get_draw_flag">
+			If [code]true[/code], the Light in the Environment has effects on the Sprite. Default value: [code]false[/code].
```
Comment (cbscribe) [inline doc/classes/SpriteBase3D.xml]: These should be links to the classes: `[Light]` and `[Environment]`

File: doc/classes/SpriteBase3D.xml [hunk 10]
```diff
@@ -164,36 +166,51 @@
 		<member name="alpha_cut" type="int" setter="set_alpha_cut_mode" getter="get_alpha_cut_mode" enum="SpriteBase3D.AlphaCutMode">
 		</member>
 		<member name="axis" type="int" setter="set_axis" getter="get_axis" enum="Vector3.Axis">
+			The Direction in wich the front of the texture faces.
```
Comment (cbscribe) [inline doc/classes/SpriteBase3D.xml]: Direction -> direction
wich -> which

File: doc/classes/Sprite.xml [hunk 11]
```diff
@@ -211,40 +211,42 @@
...
 		<member name="frame" type="int" setter="set_frame" getter="get_frame">
-			Current frame to display from sprite sheet. [code]vframes[/code] or [code]hframes[/code] must be greater than 1.
+			Current frame to display from sprite sheet. [member vframes] or [member hframes] must be greater than 1.
 		</member>
 		<member name="hframes" type="int" setter="set_hframes" getter="get_hframes">
-			The number of horizontal frames in the sprite sheet.
+			The number of collumns in the sprite sheet.
 		</member>
 		<member name="normal_map" type="Texture" setter="set_normal_map" getter="get_normal_map">
+			The normalmap gives depth to the Sprite.
```
Comment (cbscribe) [inline doc/classes/Sprite.xml]: normal map

File: doc/classes/AnimatedSprite.xml [hunk 12]
```diff
@@ -161,7 +161,7 @@
 			If [code]true[/code] texture is flipped vertically. Default value: [code]false[/code].
 		</member>
 		<member name="frame" type="int" setter="set_frame" getter="get_frame">
-			The current frame index.
+			The index of the animation frame the node displays on screen.
```
Comment (cbscribe) [inline doc/classes/AnimatedSprite.xml]: This is overwritten ("on screen" is unncessary). I liked the first one, but if you want to be more specific, try something like:

    The displayed animation frame's index.

File: doc/classes/AnimatedSprite3D.xml [hunk 13]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AnimatedSprite3D" inherits="SpriteBase3D" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Sprite node in 3D that can use multiple 2D textures for animation.
```
Comment (cbscribe) [inline doc/classes/AnimatedSprite3D.xml]: "Sprite node in 3D" -> "3D sprite node"

File: doc/classes/AnimatedSprite3D.xml [hunk 14]
```diff
@@ -83,18 +85,22 @@
 	</methods>
 	<members>
 		<member name="animation" type="String" setter="set_animation" getter="get_animation">
+			The current animation from the [code]frames[/code] resource. If this value changes, the [code]frame[/code] counter is reset.
 		</member>
 		<member name="frame" type="int" setter="set_frame" getter="get_frame">
+			The index of the animation frame the node displays on screen.
```
Comment (cbscribe) [inline doc/classes/AnimatedSprite3D.xml]: see above comment on AnimatedSprite

**Review by cbscribe — CHANGES_REQUESTED** [review]

**Review by cbscribe — APPROVED** [review]

**Pull request merged**